AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/backgroundfilter/backgroundblur/BackgroundBlurVideoFrameProcessor.swift

```diff
@@ -0,0 +1,141 @@
+        guard let foregroundMask = backgroundFilterProcessor.createForegroundAlphaMask(inputFrameCG: inputCgFrame,
+                                                                                       inputFrameCI: inputFrame)
+        else {
+            return
+        }
+
+        // Blur the input frame with Gaussian Blur which will be used as the background of the final output image.
+        let backgroundBlurredImage: CIImage = inputFrame.clampedToExtent()
+                                                        .applyingGaussianBlur(sigma: Double(blurStrength))
+                                                        .clamped(to: inputFrame.extent)
+
+        // Create the final output image by blending the alpha mask on top of the input frame to produce
+        // the foreground image which is placed on top of the blurred background image.
+        guard let outputImage: CIImage = backgroundFilterProcessor.blendWithWithAlphaMask(inputFrameCI: inputFrame,
+                                                                                          maskImage: foregroundMask,
+                                                                                          backgroundImage: backgroundBlurredImage)
+        else {
+            backgroundFilterProcessor.logger.error(msg: "Error producing the final output image.")
+            return
+        }
+
+        // Create the final `CVPixelBuffer` for the output image.
+        var mergedImageBuffer: CVPixelBuffer?
+        guard let bufferPool = backgroundFilterProcessor.getBufferPool() else {
+            backgroundFilterProcessor.logger.error(msg: "Error retrieving final buffer pool.")
+            return
+        }
+        CVPixelBufferPoolCreatePixelBuffer(nil, bufferPool, &mergedImageBuffer)
+        guard let validMergedImageBuffer = mergedImageBuffer else {
+            backgroundFilterProcessor.logger.error(msg: "Error creating CVPixelBuffer for output image.")
+            return
+        }
+
+        // Render the output CGImage image in the buffer.
+        context.render(outputImage, to: validMergedImageBuffer)
+
+        processedFrame = VideoFrame(timestampNs: frame.timestampNs,
+                                    rotation: frame.rotation,
+                                    buffer: VideoFramePixelBuffer(pixelBuffer: validMergedImageBuffer))
+    }
+
+    /// Allow builders to change the blur intensity value after initialization.
+    ///
+    /// - Parameters:
+    ///   - newBlurStrength: New blur intensity value.
+    public func setBlurStrength(newBlurStrength: BackgroundBlurStrength) {
+        blurStrength = newBlurStrength.rawValue
+    }
+
+    /// Adds a video sink to the sinks set.
+    public func addVideoSink(sink: VideoSink) {
+        if !sinks.contains(sink) {
+            sinks.add(sink)
+        }
+    }
+
+    /// Remove a video sink from the sinks set.
+    public func removeVideoSink(sink: VideoSink) {
+        if sinks.contains(sink) {
```

Comment:
ditto

Reply:
I removed this but before we removing shouldn't we check if item exists ? I couldn't find a documentation whether `nsmutableset.remove()` checks for that but I'm sure its guarded somewhere.

Reply:
It says in the documentation https://developer.apple.com/documentation/swift/set/1540257-remove. If nothing is removed, returns `nil`.